compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -850,23 +922,23 @@ private boolean analyzeStaticMatchPatterns(BLangMatch matchStmt) {
             return false;
         }
         List<BLangMatchStaticBindingPatternClause> matchedPatterns = new ArrayList<>();
-        for (BLangMatchStaticBindingPatternClause pattern : matchStmt.getStaticPatternClauses()) {
-            analyzeNode(pattern, env);
+        for (BLangMatchStaticBindingPatternClause patternClause : matchStmt.getStaticPatternClauses()) {
+            analyzeNode(patternClause, env);
 
             List<BType> matchedExpTypes = matchStmt.exprTypes
                     .stream()
-                    .filter(exprType -> isValidStaticMatchPattern(exprType, pattern.literal))
+                    .filter(exprType -> isValidStaticMatchPattern(exprType, patternClause.literal))
```

Comment:
Shall we replace the streams since we're updating this bit?

Reply:
changed.